webui/react/src/components/WorkspaceCreateModal.tsx

```diff
@@ -42,12 +47,35 @@ const CodeEditor = React.lazy(() => import('hew/CodeEditor'));
 
 const WorkspaceCreateModalComponent: React.FC<Props> = ({ onClose, workspaceId }: Props = {}) => {
   const idPrefix = useId();
-  const { canModifyWorkspaceAgentUserGroup, canModifyWorkspaceCheckpointStorage } =
-    usePermissions();
+  const {
+    canModifyWorkspaceAgentUserGroup,
+    canModifyWorkspaceCheckpointStorage,
+    canSetWorkspaceNamespaceBindings,
+  } = usePermissions();
   const [form] = Form.useForm<FormInputs>();
+  //const namespaceName = Form.useWatch('useAgentUser', form);
   const useAgentUser = Form.useWatch('useAgentUser', form);
   const useAgentGroup = Form.useWatch('useAgentGroup', form);
   const useCheckpointStorage = Form.useWatch('useCheckpointStorage', form);
+  const [resourceManagers, setResourceManagers] = useState<string[]>([]);
+  const [namespaceBindingsList, setnamespaceBindingsList] = useState<{ [key: string]: string }>({});
+  const fetchResourceManagers = useCallback(async () => {
+    try {
+      const response = await getKubernetesResourceManagers({});
+      setResourceManagers(response.names);
+    } catch (e) {
+      handleError(e, {
+        level: ErrorLevel.Error,
+        publicMessage: 'Failed to fetch Resource Managers.',
+        silent: true,
+        type: ErrorType.Server,
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    fetchResourceManagers();
+  }, [fetchResourceManagers]);
```

Comment:
please use the `useAsync` helper for this kind of state.
```suggestion
  const resourceManagers = useAsync(async (canceller) => {
    try {
      const response = await getKubernetesResourceManagers({}, { signal: canceller.signal });
      return responses.names;
    } catch (e) {
      handleError(e, {
        level: ErrorLevel.Error,
        publicMessage: 'Failed to fetch Resource Managers.',
        silent: true,
        type: ErrorType.Server,
      });
    }
  }, [])
  const namespaceBindingsList = useAsync(async (canceller) => {
    try {
      const clusterNamespacePairs = await listWorkspaceNamespaceBindings({ id: workspaceId || 0 }, { signal: canceller.signal });
      return clusterNamespacePairs.clusterNamespacePairs
    } catch (e) {
      handleError(e, {
        level: ErrorLevel.Error,
        publicMessage: 'Failed to fetch list of workspace namespace bindings.',
        silent: true,
        type: ErrorType.Server,
      });
    }
  }, [workspaceId]);
```

note that these return `Loadable` objects, which encourage you to handle the case where the requests aren't loaded yet.

Reply:
done!